src/config/cloudtrail.json

```diff
@@ -0,0 +1,753 @@
+{
+  "name": "CLOUD_TRAIL",
+  "displayName": "Cloud Trail Management Events / Logs",
+  "rules": [
+    {
+      "aws": {
+        "logContentParseAs": "json"
+      },
+      "sources": [
+        {
+          "sourceType": "logs",
+          "source": "log_group",
+          "condition": "$prefix('aws-cloudtrail-logs')"
+        }
+      ],
+      "attributes": [
+        {
+          "key": "aws.service",
+          "pattern": "'cloudtrail'"
+        },
+        {
+          "key": "audit.action",
+          "pattern": "log_content_parsed.eventName"
+        },
+        {
+          "key": "audit.identity",
+          "pattern": "log_content_parsed.userIdentity.arn"
+        },
+        {
+          "key": "audit.result",
+          "description": "If a field errorCode exists, then return Failure.{errorCode}, else return Succeeded",
+          "pattern": "if( log_content_parsed.errorCode==null, &'Succeeded', &[`Failed`, log_content_parsed.errorCode] | [? @!=null ] | join('.', @) , @)"
+        },
+        {
+          "key": "severity",
+          "pattern": "if( log_content_parsed.errorCode==null, &'INFO', &'ERROR', @)"
+        },
+        {
+          "key": "dt.source_entity",
+          "pattern": "if( log_content_parsed.eventName == 'AddRoleToDBCluster', &dt_meid_supporting_service_v2( 'Aurora', format('arn:aws:rds:{}:{}:cluster:{}', [region, account_id, log_content_parsed.requestParameters.dBClusterIdentifier])), &EmptyValue, @ )"
+        },
+        {
+          "key": "aws.arn",
+          "pattern": "if( log_content_parsed.eventName == 'AddRoleToDBCluster', &format('arn:aws:rds:{}:{}:cluster:{}', [region, account_id, log_content_parsed.requestParameters.dBClusterIdentifier]), &EmptyValue, @ )"
```

Comment:
Do I remember correctly that you wanted to try to avoid repeating the ARN calculation in `dt.source_entity` and `aws.arn`? Is it still your plan?

Reply:
I did, but I had trouble taming the chaos and had to write it all out plainly first before seeing the patterns and deciding what to do about it. I might need some follow up discussion about the options here.